bpfman: Add support for fentry/fexit program types

Billy99 opened the pull request:
Add support for the fentry and fexit program types.

Resolves: #1012

astoycos commented on bpfman/src/bpf.rs:
```diff
@@ -658,6 +664,52 @@ impl BpfManager {
+                let owned_link: FEntryLink = fentry.take_link(link_id)?;
+                let fd_link: FdLink = owned_link.into();
+                fd_link
+                    .pin(format!("{RTDIR_FS}/prog_{}_link", id))
```
```suggestion
                    .pin(format!("{RTDIR_FS}/prog_{id}_link"))
```

astoycos replied:
same below

Billy99 replied:
Done.
There are multiple places that the variable is in the string, why is this method preferred?

astoycos replied:
If it's already in the right format it's just a bit less code :) but really it's just a style thing

astoycos commented on tests/integration-test/src/tests/utils.rs:
```diff
@@ -460,6 +464,66 @@ pub fn add_kretprobe(
 }
 
+/// Install a fentry program with bpfman
+pub fn add_fentry(load_type: &LoadType, image_url: &str, file_path: &str) -> Result<String> {
```
Nit but can you just collapse these into a single function with a boolean toggle or something? We already have so much duplicated code here. :/

Billy99 replied:
Done

astoycos reviewed (approved):
Just some small nits, we'll make the integration test program a bit better in a follow up, thanks @Billy99!!!

astoycos commented:
(just remove hold when ready)

Billy99 commented on bpfman/src/bpf.rs:
```diff
@@ -337,6 +337,10 @@ impl BpfManager {
                     let _ = self.cleanup_map_pin_path(&pin_path, map_owner_id);
                 }
+
+                // Cleanup and program that failed to create. Ignore any delete errors.
```
Done

Billy99 commented:
I squashed the integration tests in with the main fentry/fexit code (because I addressed one of the comments in the wrong commit and just made it easier to combine them than to try to unwind the update).
